### DISCUSSION TITLE
v0.2.13

### DISCUSSION THREAD

**Discussion by smarthausleben** [smarthausleben/ioBroker.wamo#20] [Announcements]
<hr /><em>This discussion was created from the release <a href='https://github.com/smarthausleben/ioBroker.wamo/releases/tag/v0.2.13'>v0.2.13</a>.</em>

**Comment by weindler** [comment]
Hi, wäre es vielleicht möglich, daß man in der nächsten Version, einen Button mit einbaut, nach meinetwegen 2000Liter Durchflussmenge ein Selbsttest durchgeführt wird, mit einem Eintrag in einen neuen Datenpunkt. Selbsttest ok. (Unter Selbsttest meine ich, wenn kein Wasser fließt, Hahn schließen und nach 5 Minuten wieder öffnen. Wäre auch gut um ein Verrosten zuvorzukommen und die Gewährleistung aufrecht zu erhalten daß er auch im Notfall schließen kann. Nicht daß das Gerät ewige Zeit auf ist und dann durch Dreck und Rost sich nicht mehr schließen lässt. Was hältst du davon. Aktuell mache ich es mit einem Blockly. Es wäre natürlich schön wenn es direkt im Adapter verfügbar wäre.

**Reply by smarthausleben** [reply]
Irgendwie habe ich dein Post hier wohl übersehen, sorry.
Wenn ich dich richtig verstanden habe, möchtest du so etwas wie eine "Bewegungsfahrt", ähnlich wie bei elektronisch Heizkörperventilen, mit dem Kugelhahn realisieren um einem Festsetzen etc. zuvor zu kommen?
Wäre es nicht sinnvoller das Ganze in regelmäßigen Abständen und zu einer bestimmten Uhrzeit, z.B. Nachts, durchzuführen?

**Reply by weindler** [reply]
ja genau, so mache ich das jetzt auch, kein Problem habe ein Script dafür, ich meinte nur daß man dies noch miteinbauen könnte.

[syr script.txt](https://github.com/smarthausleben/ioBroker.wamo/files/10335163/syr.script.txt)

**Reply by smarthausleben** [reply]
Hallo weindler,

Ich bin gerade dabei die von dir gewünschte "Jam Protection" Funktionalität zu implementieren.

Ab dem nächsten Release wird es in den Adapter Settings ein weiteres Tab (TASKS) geben wo man die "Bewegungsfahrt" aktivieren und mittels CRON Zeitplan planen kann.

Ich konnte das ganze, mangels passendem Zeitpunkt, allerdings noch nicht real testen, da dabei ja ständig das Ventil geschlossen und geöffnet würde und meine Frau darüber wohl nicht sehr erfreut wäre 😂

Während der Bewegungsfahrt werden keine anderen States wie z.B. Ventilstellung, Alarme etc. aktualisiert um nicht Scrips welche eventuell darauf triggern auszulösen. Es gibt aber einen zusätzlichen State mit welchem die Bewegungsfahrt angezeigt wird.

Alle regulären Anfragen an das Gerät werden während der Bewegungsfahrt ebenfalls nicht ausgeführt und eine entsprechende Warnmeldung im Log abgesetzt. Nach dem Ende der Aktion werden die Anfragen dann wieder ganz normal ausgeführt.

Im aktuelle Code auf Github ist nur noch der eigentlich Schießbefehl deaktiviert, falls du es mal testen möchtest. Allerding werden nach dieser Nachricht sicher noch andere Änderungen folgen, also Vorsicht. Zum Zeitpunkt dieser Nachricht ist die Github Action bei #665.

Gruß Claus

**Reply by weindler** [reply]
Hi, danke dir, kein Problem, mein Script läuft ja jeden 2. Monat zuverlässig, ich warte mal ab bis es eine Version gibt die nach deinen Änderungen zum testen ist und werde dann berichten, mach dir keinen Streß, eilt nicht. Danke dir für die Entwicklung des Adapters.
